research/src/pages/popup/popup.proposal.mdx

```diff
@@ -0,0 +1,434 @@
+  </dd>
+  <dt>
+    <span data-x="concept-element-accessibility-considerations">Accessibility considerations</span>:
+  </dt>
+  <dd>
+    Pending specification in <a href="https://w3c.github.io/html-aria/">ARIA in HTML</a> and{' '}
+    <a href="https://w3c.github.io/html-aam/">HTML-AAM</a>.
+  </dd>
+</dl>
+
+<p class="todo">
+  Add IDL to frontmatter (<a href="https://github.com/openui/open-ui/issues/356">#356</a>)
+</p>
+
+A `popup` element represents transient application UI that “pops up” in the
+[top layer](https://fullscreen.spec.whatwg.org/#top-layer) of a web document, enabling the user to
+perform some task in a timely and immediate fashion. `popup` is similar to the `dialog` element, but
+is distinguished by its transient nature (“light dismiss” behaviors) and mutual exclusivity: only
+one `popup` element may be rendered at a time, with the exception of nested `popup`s.
+
+When a `popup` is
+[browsing-context connected](https://html.spec.whatwg.org/#becomes-browsing-context-connected), set
+its _open state_ to `false`, prior to running any
+[showing a `popup` element steps](#showing-popup-steps) on the element, as applicable.
+
+<p class="note">
+  Examples of popups include: button menus, listboxes used for item selection e.g. comboboxes,
+  "teaching" UI, and other progressive-disclosure widgets or content pickers.
+</p>
+
+## Showing a `popup` element
+
+A `popup` element and its contents should not be rendered unless:
+
+- The `initiallyopen` attribute is specified
+- The `show()` method is invoked
+- Or the `popup` element has been invoked through user interaction with an element whose `popup`
+  attribute references the `popup` element's ID
+
+This requirement may be implemented indirectly through the style layer. For example,
+user agents that
+[support the suggested default rendering](https://html.spec.whatwg.org/multipage/infrastructure.html#renderingUA)
+implement this requirement using the CSS rules described in the
+[Rendering section](https://html.spec.whatwg.org/multipage/rendering.html#rendering).
+
+<h3 id="popup-stack">The popup stack</h3>
+
+This specification introduces a new stacking layer in the document
+[top layer](https://www.w3.org/TR/CSS2/zindex.html), called the **popup stack**. Each document has
+one popup stack.
+
+The popup stack is an [ordered set](https://infra.spec.whatwg.org/#ordered-set) of elements,
+rendered in the order they appear in the set. The last element in the set is rendered last,
+and thus appears on top.
+
+Showing or hiding a `popup` element adds or removes the element from the popup stack, respectively.
+
+### The `initiallyopen` attribute
+
+The `initiallyopen` attribute is a
+[boolean attribute](https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#boolean-attribute).
+When specified, it indicates that the `popup` element should have the `show()` method called upon
+document load, such that the user can interact with it.
```

Comment:
Thanks, I'm not sure why I missed these shoulds. Perhaps because not written as SHOULD? :)

Anyway, addressed in 5fc5041409d28c395e81cf30f039a7c0e269b7fe